server.js

```diff
@@ -223,10 +224,9 @@ if (process.env.DEBUG_DUMMY_SMTP) {
   }
 }
 
-if (gTransporter) {
-  app.listen(port, function() {
-    console.log("Listening on " + port);
-  });
-} else {
-  console.error("SMTP credentials couldn't be read from the environment, exiting.");
+if (!gTransporter) {
+  console.log("SMTP config unavailble. Email features will not work.")
 }
+app.listen(port, function() {
```

Comment:
Let's just not run the server if we failed to get valid SMTP config, for now. In the future we can account for invalid config in the APIs handlers that use SMTP.